plugins/besecure.rb

```diff
@@ -254,96 +205,146 @@ def cmd_besecure_report_download(*args)
           print_error("#{@hostname} - Unable to parse the response")
           return ''
         end
-    
+
         if body['error']
           print_error("#{@hostname} - An error occured:")
           print_error(body)
           return ''
         end
 
-        decompressed = ''
-        if format_file != 'json'
-          compressed_base64 = body["compresseddata"]
-          compressed = Base64.decode64(compressed_base64)
-          decompressed = Zlib::Inflate.inflate(compressed)
-        else
-          decompressed = body
+        data = body['data']
+        data.each do |item|
+          tbl << [ item['ID'], item['Name'], item['PrettyRange']]
         end
 
-        if @debug
-          print_status(decompressed)
-        end
+        # print_good(body)
 
-        ::FileUtils.mkdir_p(args[2])
-        name = ::File.join(args[2], args[3])
-        print_status("Saving report to #{name}")
-        output = ::File.new(name, "w")
-        output.puts(decompressed)
-        output.close
-       
-        ###
-        # Return the report
-        return decompressed
-      else
-        print_status("Usage: besecure_report_download <network_id> <format_name> <path> <report_name>")
+        print_good('beSECURE list of reports')
+        print_line
+        print_line tbl.to_s
+        print_line
       end
-      
-      return ''
-    end
 
-    def cmd_besecure_report_import(*args)
-      if args?(args, 2)
-        if !database?
-          print_error("Database not ready")
-          return ''
+      def cmd_besecure_report_download(*args)
+        if args?(args, 4)
+          req = Net::HTTP::Post.new('/json.cgi', initheader: { 'Host' => @hostname })
```

Comment:
```suggestion
          req = Net::HTTP::Post.new('/json.cgi', { 'Host' => @hostname })
```

Crash:

``2.7.2 :021 > req = Net::HTTP::Post.new("/", initheader: { 'Host' => 'helloworld' }); http = Net::HTTP.new('127.0.0.1', 5000); http.set_debug_output($stdout); http.request(req)
Traceback (most recent call last):
	27: from /Users/user/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `<main>'
	26: from /Users/user/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `eval'
	25: from /Users/user/.rvm/gems/ruby-2.7.2/bin/irb:23:in `<main>'
	24: from /Users/user/.rvm/gems/ruby-2.7.2/bin/irb:23:in `load'
	23: from /Users/user/.rvm/gems/ruby-2.7.2/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
	 6: from (irb):21:in `<main>'
	 5: from (irb):21:in `new'
	 4: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/request.rb:15:in `initialize'
	 3: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/generic_request.rb:46:in `initialize'
	 2: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:16:in `initialize_http_header'
	 1: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:16:in `each'
/Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:21:in `block in initialize_http_header': undefined method `strip' for {"Host"=>"helloworld"}:Hash (NoMethodError)`
```

No crash:

```
2.7.2 :022 > req = Net::HTTP::Post.new("/", { 'Host' => 'helloworld' }); http = Net::HTTP.new('127.0.0.1', 5000); http.set_debug_output($stdout); http.request(req)
opening connection to 127.0.0.1:5000...
opened                                                                    
<- "POST / HTTP/1.1\r\nHost: helloworld\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nConnection: close\r\nContent-Length: 0\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\n"                 
<- ""
```

And headers verified via ncat:

```ncat -lnvp 5000
Ncat: Version 7.93 ( https://nmap.org/ncat )
Ncat: Listening on :::5000
Ncat: Listening on 0.0.0.0:5000
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49718.
POST / HTTP/1.1
Host: helloworld
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
Accept: */*
User-Agent: Ruby
Connection: close
Content-Length: 0
Content-Type: application/x-www-form-urlencoded


```

Reply:
`initheader=` was changed to `initheader:` in 3 locations in the besure plugin. Presumably they should all be removed.

Reply:
Good call; Updating the other locations sounds good :+1:

Reply:
Replaced